os/fs/driver/mtd/smart.c

```diff
@@ -1875,36 +1797,16 @@ static int smart_scan(FAR struct smart_struct_s *dev)
 		}
 
 		/* Test if this sector has been committed. */
-
-		if (!status_committed) {
```

Comment:
- Checks to tackle specific error cases should be avoided.
- General checks should handle all cases.